### DISCUSSION TITLE
How can I edit a GGML model?

### DISCUSSION THREAD

**Discussion by cpt9m0** [ggml-org/llama.cpp#2139] [General]
Howdy!

How can I edit/change/update, or remove/disable the weights in a converted-to-ggml model?

**Comment by KerfuffleV2** [comment]
You'd have to write your own tool/script for it. The good news is that the GGML format is relatively simple. Doing stuff like removing a weight (I assume you mean tensors) would require rewriting the whole file though. Also, the model architecture expects specific weights with specific names, so removing one will probably make it so you can't load that model.

You could _possibly_ do something like overwrite one specific tensor with different data but unless it was exactly the same dimensions/quantization then you'd still have to rewrite the whole file.

For the most part, just reconverting it is going to be easier than trying to mess with existing GGML files.
